python_modules/dagster/dagster_tests/daemon_tests/test_backfill.py

```diff
@@ -570,10 +592,12 @@ def test_canceled_backfill(
     assert instance.get_runs_count() == 1
 
 
+@pytest.mark.parametrize("parallel", [True, False])
```

Comment:
this pattern makes sense to me. I think we just want a couple of additional test cases here to make sure that this is being exercised fully:

- something that touches an asset backfill with the threadpool in addition to a job backfill (ideally one that requires more than one iteration, e.g. needing to backfill a parent asset before backfilling a child asset - something that would have caught the futures issue from before where the second iteration fought with the first iteration). There are a couple of examples in this same file that touch asset backfills, e.g. test_asset_backfill_from_asset_graph_subset

- something that does two backfills in parallel at once.